bioinformatics-utils/main_dna_rna_filter.py

```diff
@@ -0,0 +1,118 @@
+from utils.dna_rna_helper import (
+    transcribe, reverse, complement, reverse_complement, gc_content,
+    is_palindrome, melting_temperature, change_sequence
+)
+from utils.filter_helper import avg_quality, read_fastq as fastq_to_dict, \
+    write_fastq, gc_content as filter_gc_content
+from typing import Any, Union
+
+
+def filter_fastq(
+    input_fastq: str,
+    output_fastq: str,
+    gc_bounds: tuple[float, float] = (0, 100),
+    length_bounds: tuple[int, int] = (0, 2**32),
+    quality_threshold: float = 0
+) -> None:
+    """
+    Filters reads based on GC content, sequence length, and average quality,
+    and writes filtered reads to a file.
+
+    Arguments:
+    - input_fastq: Path to the input FASTQ file.
+    - output_fastq: Path to the output FASTQ file.
+    - gc_bounds: Range for the percentage of GC content (default is (0, 100)).
+    - length_bounds: Range for the sequence length (default is (0, 2**32)).
+    - quality_threshold: Threshold for the average quality of the read
+      (default is 0).
+
+    Returns:
+    - None. Writes the filtered sequences to the output FASTQ file.
+    """
+
+    seqs = fastq_to_dict(input_fastq)
+
+    if isinstance(gc_bounds, (int, float)):
+        gc_bounds = (0, gc_bounds)
+    if isinstance(length_bounds, (int, float)):
+        length_bounds = (0, length_bounds)
+
+    filtered_seqs = {}
+
+    for name, (seq, quality) in seqs.items():
+        gc = filter_gc_content(seq)
+        length = len(seq)
+        avg_qual = avg_quality(quality)
+
+        if (gc_bounds[0] <= gc <= gc_bounds[1] and
+                length_bounds[0] <= length <= length_bounds[1] and
+                avg_qual >= quality_threshold):
+            filtered_seqs[name] = (seq, quality)
+
+    write_fastq(output_fastq, filtered_seqs)
+
+
+def run_dna_rna_tools(*args: str, **kwargs: Any) -> Union[str, list[str]]:
+    """
+    Performs various DNA/RNA sequence operations such as transcription,
+    complement, reverse complement, and more.
+
+    Arguments:
+    - *args: The DNA/RNA sequences and the operation to perform.
+      The last argument must be the operation name (str).
+        - Example operations: "transcribe", "complement", "reverse_complement",
+          "gc_content", "is_palindrome", "melting_temperature",
+          "change_sequence", "reverse".
+    - **kwargs: Additional keyword arguments to customize operations
+      (e.g., "case" for case transformation in `change_sequence`).
+
+    Returns:
+    - A string result of the operation for a single sequence or a list of
+      results for multiple sequences. If the sequences contain invalid
+      characters, an error message will be returned.
+    """
+
+    def validate_seq(seq: str) -> str:
```

Comment:
В идеальном случае функция возвращает None.
В связи с этим стоит наверное добавить 
```suggestion
    def validate_seq(seq: str) -> None|str:
```